src/Extensions/Abstractions/MachineInformation/EmptyMachineInformation.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using System.Reflection;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>Base class for machine information</summary>
+	public class EmptyMachineInformation : IMachineInformation
+	{
+		/// <summary>Create instance of machine information</summary>
+		public EmptyMachineInformation()
+		{
+			MachineName = DefaultEmptyValue;
+			MachineId = DefaultEmptyValue;
+			MachineRole = DefaultEmptyValue;
+			MachineClusterIpAddress = IPAddress.None;
+			MachineCount = 1;
+			MachineCluster = MachineId;
+
+			EnvironmentName = DefaultEmptyValue;
+			DeploymentSlice = DefaultEmptyValue;
+
+			RegionName = DefaultEmptyValue;
+			ServiceName = DefaultEmptyValue;
+			BuildVersion = DefaultEmptyValue;
+
+			IsCanary = false;
+			IsPrivateDeployment = false;
+		}
+
+
+		/// <inheritdoc/>
+		public string MachineId { get; protected set; }
+
+
+		/// <inheritdoc/>
+		public string MachineName { get; protected set; }
```

Comment:
Consider making almost all of them readonly fields

Reply:
I need to change those values in the constructor and sometimes methods :( of types that will extend this one